### PR TITLE
gh-pages refactor

### PR DESCRIPTION
**Pull request by rstacruz**
Does stuff outlined in #1714.

No content changes, just reorganizing things.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,10 +1,11 @@
...
 
-clean:
-	rm -f index.html
+build: bundle
+	${bundle} exec jekyll build --safe
 
-.PHONY: clean
\ No newline at end of file
+bundle:
+	${bundle}
```
Comment (rstacruz) [inline Makefile]: I've rewritten the makefile to be a helper to start jekyll. This isn't really necessary, but it's convenient.

```sh
make
#=> bundle exec jekyll serve --safe --drafts --wach --port 3000
```

File: _layouts/default.html [hunk 2]
```diff
@@ -3,22 +3,26 @@
   <head>
     <meta charset="utf-8">
     <title>Mocha - the fun, simple, flexible JavaScript test framework</title>
-    <link rel="stylesheet" href="normalize.min.css">
-    <link rel="stylesheet" href="style.css">
+    <link rel="stylesheet" href="assets/normalize.min.css">
+    <link rel="stylesheet" href="assets/style.css">
     <link rel="shortcut icon" href="favicon.ico">
     <script src="http://ajax.googleapis.com/ajax/libs/jquery/1.7.0/jquery.min.js"></script>
-    <script>window.jQuery || document.write('<script src="jquery.js"><\/script>')</script>
```
Comment (rstacruz) [inline _layouts/default.html]: I've removed jquery.js from the repo just to simplify things... lemme know if you want it back.

File: index.md [hunk 3]
```diff
@@ -67,8 +70,8 @@ Mocha is a feature-rich JavaScript test framework running on [node.js](http://no
 
     $ npm install -g mocha
 
-<h2 id="getting-started">1. 2. 3. Mocha!</h2>
-
+## 1. 2. 3. Mocha!
+{:#getting-started}
```
Comment (rstacruz) [inline index.md]: I've changed the h2/h3 headings to have `kramdown`-style ID's for simplicity. Also technically not necessary, but pretty convenient...

File: _layouts/default.html [hunk 4]
```diff
@@ -3,22 +3,26 @@
...
+    <!--[if lt IE 9]><script src='//cdnjs.cloudflare.com/ajax/libs/html5shiv/3.7.2/html5shiv.min.js'></script><![endif]-->
   </head>
   <body>
     <section id="content">
       <h1><a href="https://github.com/mochajs/mocha">Mocha</a></h1>
       <p id="tag"><em>simple</em>, <em>flexible</em>, <em>fun</em></p>
+      {{content}}
+    </section>
+    <footer>
+      <span>© 2011 TJ Holowaychuk. All rights reserved.</span>
```
Comment (rstacruz) [inline _layouts/default.html]: I've left this as is... might wanna change that sometime.

File: _layouts/default.html [hunk 5]
```diff
@@ -3,22 +3,26 @@
...
+    <script src="assets/highlight.js"></script>
     <script>
       setTimeout(function(){
         $('body').addClass('onload');
       }, 0);
     </script>
-    <!--[if lt IE 9]>
-    <script src="html5shiv.js"></script>
-    <![endif]-->
+    <!--[if lt IE 9]><script src='//cdnjs.cloudflare.com/ajax/libs/html5shiv/3.7.2/html5shiv.min.js'></script><![endif]-->
```
Comment (rstacruz) [inline _layouts/default.html]: Fetch html5shiv from the CDN because who cares about html5shiv.

File: assets/highlight.js [hunk 6]
```diff
@@ -1,10 +1,21 @@
-
-$(function(){
-  $('code').each(function(){
-    $(this).html(highlight($(this).text()));
-  });
```
Comment (rstacruz) [inline assets/highlight.js]: I've removed the jQuery dependency too. Makes no sense to load jQuery simply for `$('code').each` and `$(document).ready`.

File: _layouts/default.html [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+    <link rel="stylesheet" href="assets/normalize.min.css">
+    <link rel="stylesheet" href="assets/style.css">
+    <link rel="shortcut icon" href="favicon.ico">
+    <script src="assets/highlight.js"></script>
+    <script>
+      setTimeout(function(){
+        document.body.className += ' onload';
+      }, 0);
+    </script>
+    <!--[if lt IE 9]><script src='//cdnjs.cloudflare.com/ajax/libs/html5shiv/3.7.2/html5shiv.min.js'></script><![endif]-->
```
Comment (rstacruz) [inline _layouts/default.html]: Now fetching html5shiv via cdn because who cares about legacy IE?

File: _layouts/default.html [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+  </head>
+  <body>
+    <section id="content">
+      <h1><a href="https://github.com/mochajs/mocha">Mocha</a></h1>
+      <p id="tag"><em>simple</em>, <em>flexible</em>, <em>fun</em></p>
+      {{content}}
+    </section>
+    <footer>
+      <span>© 2011 TJ Holowaychuk. All rights reserved.</span>
```
Comment (rstacruz) [inline _layouts/default.html]: I've left this as is, since this PR is only about restructuring things (not updating content). Might wanna change this.

**Comment by rstacruz** [comment]
btw this is an alternate implementation to #1726... only:

- no actual content changes. grammar and stuff are left alone.
- no marked-toc
- syntax highlighting is still client-side

**Comment by rstacruz** [comment]
demo for now: http://ricostacruz.com/mocha

**Comment by jbnicolai** [comment]
I won't have time to look at this in more care until later tonight, but big :+1:! The css issue in #1726 seems to not be present on your demo (which is great), and no changes to grammar etc only helps in making it more modular and easy to merge.

> toc looks exactly like how it is now, not the nested version in #1726

No worries, I believe this is a bug due to `style.css` somehow not being applied (even though it's loaded, haven't figured out the root cause yet), so you're version is right :)

**Comment by boneskull** [comment]
@jbnicolai what's the CSS issue in #1726?

**Comment by boneskull** [comment]
to note, this is fully incompatible with #1726, so it's one or the other

**Comment by rstacruz** [comment]
just a subtle nudge... any updates? :)

**Comment by danielstjules** [comment]
I like the idea of merging this, and later pulling copy/doc changes from https://github.com/boneskull/mocha/blob/jekyll/index.md

For example, I like that this PR doesn't change the table of contents. For comparison:
http://boneskull.com/mocha/#table-of-contents
http://ricostacruz.com/mocha/#table-of-contents

**Comment by boneskull** [comment]
@danielstjules I don't think anybody's going to take the time to extract my copy changes.  honestly I didn't expect anyone else to submit a PR for this, which is why I did it the way I did.

The current ToC is missing stuff and needs to be maintained manually.

File: Makefile (hunk 1 above)
Comment (jbnicolai) [inline Makefile]: :+1:

File: _layouts/default.html (hunk 8 above)
Comment (jbnicolai) [inline _layouts/default.html]: :+1: gotcha

**Comment by boneskull** [comment]
@rstacruz going to close this in lieu of #1777, please continue conversation there

**Pull request closed without merging**